src/main/java/com/notably/model/BlockModelImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockImpl;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeImpl;
+import com.notably.model.block.Body;
+import com.notably.model.block.exceptions.CannotModifyRootException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+import javafx.beans.property.Property;
+
+/**
+ * The implementation class of BlockModel.
+ */
+public class BlockModelImpl implements BlockModel {
+    private BlockTree blockTree;
+    private Property<AbsolutePath> currentlyOpenPath;
+
+    public BlockModelImpl() {
+        blockTree = new BlockTreeImpl();
+    }
+
+    @Override
+    public BlockTree getBlockTree() {
+        return blockTree;
+    }
+
+    @Override
+    public AbsolutePath getCurrentlyOpenPath() {
+        return currentlyOpenPath.getValue();
+    }
+
+    @Override
+    public Property<AbsolutePath> getCurrentlyOpenPathProperty() {
+        return currentlyOpenPath;
+    }
+
+    @Override
+    public boolean hasPath(AbsolutePath p) {
+        try {
+            blockTree.get(p);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
```

Comment:
Not sure if this is a good idea tbh. But I guess it works!

Reply:
I'm trying to think of a better way, without having to use blockTreeItem methods. But this is what I came up with for now. :/